src/ripple/app/ledger/LedgerToJson.h

```diff
@@ -129,49 +132,84 @@ void fillJson (Object& json, LedgerFill const& fill)
             {
                 if (type == SHAMapTreeNode::tnTRANSACTION_NM)
                 {
-                    SerializerIterator sit (item->peekSerializer ());
-                    STTx txn (sit);
-                    txns.append (txn.getJson (0));
+                     if (bBinary)
+                    {
+                        auto&& obj = appendObject (txns);
+                        obj[jss::tx_blob] = strHex (item->peekData ());
+                    }
```

Comment:
There needs to be a destructor going off - we need to know when we finish using the subelement, so we can write the closing part of it.

Since we're streaming things as we go, we can't start a new collection before we've finished an old one.